docs/architecture/adr-042-group-module.md

```diff
@@ -0,0 +1,375 @@
+        RESULT_UNSPECIFIED = 0 [(gogoproto.enumvalue_customname) = "ProposalResultInvalid"];
+        
+        // Until a final tally has happened the status is unfinalized
+        RESULT_UNFINALIZED = 1 [(gogoproto.enumvalue_customname) = "ProposalResultUnfinalized"];
+       
+        // Final result of the tally
+        RESULT_ACCEPTED = 2 [(gogoproto.enumvalue_customname) = "ProposalResultAccepted"];
+        
+        // Final result of the tally
+        RESULT_REJECTED = 3 [(gogoproto.enumvalue_customname) = "ProposalResultRejected"];
+    }
+
+    // result is the final result based on the votes and election rule. Initial value is unfinalized.
+    // The result is persisted so that clients can always rely on this state and not have to replicate the logic.
+    Result result = 9;
+
+    // vote_state contains the sums of all weighted votes for this proposal.
+    Tally vote_state = 10 [(gogoproto.nullable) = false];
+
+    // timeout is the timestamp of the block where the proposal execution times out. Header times of the votes and execution messages
+    // must be before this end time to be included in the election. After the timeout timestamp the proposal can not be
+    // executed anymore and should be considered pending delete.
+    google.protobuf.Timestamp timeout = 11 [(gogoproto.nullable) = false];
+
+    // ExecutorResult defines types of proposal executor results.
+    enum ExecutorResult {
+        option (gogoproto.goproto_enum_prefix) = false;
+        
+        // An empty value is not allowed.
+        EXECUTOR_RESULT_UNSPECIFIED = 0  [(gogoproto.enumvalue_customname) = "ProposalExecutorResultInvalid"];
+        
+        // We have not yet run the executor.
+        EXECUTOR_RESULT_NOT_RUN = 1 [(gogoproto.enumvalue_customname) = "ProposalExecutorResultNotRun"];
+        
+        // The executor was successful and proposed action updated state.
+        EXECUTOR_RESULT_SUCCESS = 2 [(gogoproto.enumvalue_customname) = "ProposalExecutorResultSuccess"];
+        
+        // The executor returned an error and proposed action didn't update state.
+        EXECUTOR_RESULT_FAILURE = 3 [(gogoproto.enumvalue_customname) = "ProposalExecutorResultFailure"];
+    }
+
+    // executor_result is the final result based on the votes and election rule. Initial value is NotRun.
+    ExecutorResult executor_result = 12;
+
+    // msgs is a list of Msgs that will be executed if the proposal passes.
+    repeated google.protobuf.Any msgs = 13;
+}
+
+// Tally represents the sum of weighted votes.
+message Tally {
+    option (gogoproto.goproto_getters) = false;
+
+    // yes_count is the weighted sum of yes votes.
+    string yes_count = 1;
+    
+    // no_count is the weighted sum of no votes.
+    string no_count = 2;
+    
+    // abstain_count is the weighted sum of abstainers.
+    string abstain_count = 3;
+    
+    // veto_count is the weighted sum of vetoes.
+    string veto_count = 4;
+}
+```
```

Comment:
I think it would really open up governance possibilities if we supported non-binary choices or at least weren't fixed with the choices that someone could make. I'm thinking of something along the lines of:
```proto
message Tally {
    // schema represents how the choices should be interpreted 
    // (maybe this goes in the proposal or is somehow known by the decision policy)
    int64 schema = 1; 

    // choices is ordered identically to the members in the group so as 
    // to easily match the member weighting with the choice
    repeated Choice choices = 2; 
}
```
Where a `Choice` is just:
```go
type Choice uint64
```

Reply:
Yeah, we should think of ways to make it more flexible. My thinking was that yes, no, veto and abstain is a fairly common pattern for binary voting and a good starting point. But it would be cool if we supported other patterns like ranked choice, etc.

Reply:
added to the list of "Further Discussions"